MockCollector test bug

## Requirement - what kind of business use case are you trying to solve?

On my desktop (Arch Linux/Manjaro), I consistently get the following test error:
```
Error Trace:	mock_collector_test.go:162
			Error:      	An error is expected but got nil.
			Test:       	TestMockTCollectorErrors
			Messages:   	error because of bad address
```

## Problem - what in Jaeger blocks you from solving the requirement?

The test expects binding to 127.0.0 to fail. After some digging with strace I discovered that somehow the system extends the supposedly broken address of 127.0.0 to be 127.0.0.0. Then, strace shows this:
`bind(3, {sa_family=AF_INET, sin_port=htons(0), sin_addr=inet_addr("127.0.0.0")}, 16) = 0`.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Remove the assertion that requires this bind to fail seeing as it is not portable/correct.

## Any open questions to address

Find out why this is OK in Travis but not my version of Go (1.10.3) and Linux (Linux 4.18.0-1-MANJARO #1 SMP PREEMPT x86_64 GNU/Linux).

Update:

Related failures.
```
--- FAIL: TestNewZipkinThriftUDPClient (0.00s)
	assertions.go:256: 
			Error Trace:	thriftudp_client_test.go:27
			Error:      	An error is expected but got nil.
			Test:       	TestNewZipkinThriftUDPClient
=== RUN   TestNewJaegerThriftUDPClient
--- FAIL: TestNewJaegerThriftUDPClient (0.00s)
	assertions.go:256: 
			Error Trace:	thriftudp_client_test.go:38
			Error:      	An error is expected but got nil.
			Test:       	TestNewJaegerThriftUDPClient
FAIL
FAIL	github.com/jaegertracing/jaeger/cmd/agent/app/testutils	0.104s
```

--- a/cmd/agent/app/testutils/mock_collector_test.go
+++ b/cmd/agent/app/testutils/mock_collector_test.go
@@ -158,6 +158,6 @@ func TestMockTCollectorErrors(t *testing.T) {
 	_, err := startMockTCollector("", "127.0.0.1:0")
 	assert.Error(t, err, "error because of empty service name")
 
-	_, err = startMockTCollector("test", "127.0.0:0")
+	_, err = startMockTCollector("test", "256.0.0:0")
 	assert.Error(t, err, "error because of bad address")
 }

--- a/cmd/agent/app/testutils/thriftudp_client_test.go
+++ b/cmd/agent/app/testutils/thriftudp_client_test.go
@@ -23,7 +23,7 @@ import (
 )
 
 func TestNewZipkinThriftUDPClient(t *testing.T) {
-	_, _, err := NewZipkinThriftUDPClient("1.2.3:0")
+	_, _, err := NewZipkinThriftUDPClient("256.2.3:0")
 	assert.Error(t, err)
 
 	_, cl, err := NewZipkinThriftUDPClient("127.0.0.1:12345")
@@ -34,7 +34,7 @@ func TestNewZipkinThriftUDPClient(t *testing.T) {
 func TestNewJaegerThriftUDPClient(t *testing.T) {
 	compactFactory := thrift.NewTCompactProtocolFactory()
 
-	_, _, err := NewJaegerThriftUDPClient("1.2.3:0", compactFactory)
+	_, _, err := NewJaegerThriftUDPClient("256.2.3:0", compactFactory)
 	assert.Error(t, err)
 
 	_, cl, err := NewJaegerThriftUDPClient("127.0.0.1:12345", compactFactory)

